src/components/errorMessage/ErrorMessage.tsx

```diff
@@ -0,0 +1,28 @@
+import Text from 'src/components/text/Text';
```

Comment:
nitpicky about this, sorry.

omit children, we don't want that because that omits the purpose of universal components and instead creates wrappers for static components. 

what I want is ErrorMessage as the only error component:

```
import Button from "src/components/buttons/Button";
import LinkButton from "src/components/linkButton/LinkButton";
import Text from "src/components/text/Text";
import { ILink } from "studio/lib/payloads/navigation";
import styles from "./errorNews.module.css";

// custom because ErrorMessage exists natively
interface CustomErrorMessageProps {
  title: string;
  description: string;
  onClick?: () => void;
  link?: ILink;
}

const CustomErrorMessage = ({
  title,
  description,
  onClick,
  link,
}: CustomErrorMessageProps) => {
  return (
    <section className={styles.wrapper}>
      <div className={styles.error}>
        <Text type="h3">{title}</Text>
        <Text>{description}</Text>
        <div className={styles.buttonWrapper}>
          {onClick && (
            <Button type="secondary" onClick={onClick}>
              Try again
            </Button>
          )}
          {link && <LinkButton type="secondary" link={link} />}
        </div>
      </div>
    </section>
  );
};

export default CustomErrorMessage;
```

 
 that way we can replace NewsError like this in News.tsx:
```
   if (!initialPosts || posts.error) {
    const link = {
      _key: "return-home",
      _type: "link",
      linkTitle: "Return to home",
      linkType: LinkType.Internal,
      internalLink: {
        _ref: "/",
      },
    };

    return (
      <CustomErrorMessage
        title={`Error - Can’t fetch news`}
        description="Struggling to fetch stories, and we dont know why. Please be patient
          and try to refresh the page."
        onClick={fetchPosts}
        link={link}
      />
    );
  }
```


and MissingContentErrorMessage in (main)/page.tsx would be:

```
const studioLink = {
  _key: "go-to-studio",
  _type: "link",
  linkTitle: "Go to studio",
  linkType: LinkType.Internal,
  internalLink: {
    _ref: "/studio",
  },
};

const Home = async () => {
  const { perspective, isDraftMode } = getDraftModeInfo();

  const { data: landingId } = await loadQuery<string>(
    LANDING_QUERY,
    {},
    { perspective }
  );

  if (!landingId) {
    return (
      <CustomErrorMessage
        title="Landing Page Not Set"
        description="It looks like no page has been designated as your landing page in the Studio. Head over to the Studio to set a landing page and guide visitors to the right place!"
        link={studioLink}
      />
    );
  }

  const initialLandingPage = await loadQuery<PageBuilder>(
    PAGE_QUERY,
    { id: landingId },
    { perspective }
  );

  if (!initialLandingPage) {
    return (
      <CustomErrorMessage
        title="Landing Page is Missing Content"
        description={`Your landing page is set, but it looks like there’s no content yet. Visit the Studio to start adding content and make your landing page come to life!`}
        link={studioLink}
      />
    );
  }
///



```

Reply:
We can omit children. That would only limit what we can put into the error message, which is probably fine for this use case.

However, having only one component would require duplication of common props within groups of errors. For example the studio link and text strings that should be common for all errors relating to missing content. That might not be a big problem for this specific component, so I can change it to one component for now, but I think it would be limiting if applied as a general rule of the codebase.

Reply:
I disagree. If you need to duplicate link or string you can keep these as export constants in CustomErrorMessage.tsx

What you can also do is create a type for CustomErrorMessage: `type ErrorType = "missing content" | "default"` to be used in place of link and onClick in cases such as missing content. 

⚠️🚨  Ops, I've also forgot to add: `role="alert" aria-live="assertive"` to the error message for a11y ⚠️🚨